app/shared/types/templateSchema.ts

```diff
@@ -155,12 +155,70 @@ ajv.addKeyword('cantDeleteInheritedProperties', {
   },
 });
 
+ajv.addKeyword('cantReuseNameWithDifferentType', {
+  async: true,
+  errors: true,
+  type: 'object',
+  async validate(_schema: any, template: TemplateSchema) {
+    if (!template.properties || template.properties.length === 0) {
+      return true;
+    }
+
+    const condition = template.properties.map(property => ({
+      $and: [
+        {
+          name: property.name,
+          $or: [
+            { content: { $ne: property.content } },
+            { type: { $ne: property.type } },
+            { relationtype: { $ne: property.relationType } },
+          ],
+        },
+      ],
+    }));
+    const query = { properties: { $elemMatch: { $or: [...condition] } } };
+    const matchedTemplates = await model.get(query);
+    const otherTemplates = matchedTemplates.filter(t => !t._id.equals(template._id));
+    if (otherTemplates.length > 0) {
+      const errorProperties = template.properties.reduce((propertyNames: string[], property) => {
+        otherTemplates.forEach((t: TemplateSchema) => {
+          const matches = t.properties?.find(
+            p =>
+              p.name === property.name &&
+              (p.content !== property.content ||
+                p.type !== property.type ||
+                p.relationtype !== property.relationType)
+          );
+          if (matches && !propertyNames.includes(property?.name || '')) {
+            propertyNames.push(property?.name || '');
+          }
+        });
+        return propertyNames;
+      }, []);
```

Comment:
@mfacar have you considered solving this by concat every template property into one array ? or even on a map indexed by type+content,  this way there will be 2 nested iterations instead of 3 and i think it will result in a more readable implementation by default.